### ISSUE TITLE
Linking error USE_GSL=OFF

### ISSUE THREAD

**Issue by ksmith0** [pixie16/paass#82]
Built dev branch on takk (gcc 4.8.5) with following cmake flags

```
BUILD_UTKSCAN=ON
USE_GSL=OFF
```

gives this error:

```
Scanning dependencies of target utkscan
Linking CXX executable utkscan
core/CMakeFiles/CoreObjects.dir/DetectorDriver.cpp.o: In function `DetectorDriver::LoadProcessors(Messenger&)':
DetectorDriver.cpp:(.text+0x655f): undefined reference to `FittingAnalyzer::FittingAnalyzer(std::string const&)'
collect2: error: ld returned 1 exit status
make[2]: *** [Scan/utkscan/source/utkscan] Error 1
make[1]: *** [Scan/utkscan/source/CMakeFiles/utkscan.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by ksmith0** [comment]
Same issue on my laptop gcc 5.3.1-6.

No issue compiling on my laptop with `USE_GSL=ON`

**Comment by ksmith0** [comment]
Looks like `FittingAnalyzer` was declared but not defined for the situation when there is no GSL. We either need a (additonal?) class that works without GSL or a (additional?) dummy class that just says its disabled.

**Comment by spaulaus** [comment]
There is an abstracted Fit class, we can add a dummy module. One should never define the fitting analyzer without a Fitter defined, but this current implementation could cause issues.

**Comment by ksmith0** [comment]
Easy solution is to wrap DetectorDriver.cpp line 264 in precompiler `#ifdef`? 

``` c++
...
    } else if (name == "FittingAnalyzer") {
#ifdef GSL
            string type = analyzer.attribute("type").as_string();
            vecAnalyzer.push_back(new FittingAnalyzer(type));
#else
        std::cout << "No fitter defined!\n";
#endif
    } else {
...
```

**Comment by spaulaus** [comment]
No, that eliminates the versatility of the abstracted fitter class. The modification will need to be in the FittingAnalyzer directly.

**Comment by ksmith0** [comment]
Agreed that your suggestion is the correct way. Just offering a simple quick fix.

**Comment by spaulaus** [comment]
I will get to this next week. Is there a rush on this issue? Just undefine the FittingAnalyzer in the config.

**Comment by ksmith0** [comment]
No rush.

**Comment by ksmith0** [comment]
Also, it is not at the config step. It is at the compile step.

**Comment by spaulaus** [comment]
Gotcha. I will look into it.

**Comment by ksmith0** [comment]
As utkscan requires gsl for most analysis we will ignore this for the moment and downgrade it to a low priority.

**Comment by ksmith0** [comment]
I suggest we change this to `USE_GSL=ON` by default and if the user switches it to off they get an error message that it is not supported with it off.

**Comment by spaulaus** [comment]
I like that suggestion. It beats what I had originally intended to do.

**Comment by spaulaus** [comment]
I have made the USE_GSL flag dependent on the BUILD_UTKSCAN flag. USE_GSL will automatically be set to ON if BUILD_UTKSCAN is set to ON. It is false otherwise.

**Comment by spaulaus** [comment]
Resolved as of commit 456b9ca77a10f910bd779c767b9ffde94d96cc52
